src/app/pages/wallets/wallet-details/transaction-editor/pages-wallet-transaction-editor.component.ts

```diff
@@ -105,4 +117,16 @@ export class PagesWalletTransactionEditorComponent implements OnInit {
       return { amountIsZero: true };
     };
   }
+
+  private getCategoriesByType(typeToFilter: WalletTransactionType): void {
+    this.transactionCategoriesService.getCategories().subscribe((categoriesReceived) => {
+      this.transactionsCategories = categoriesReceived.filter(cat => cat.type == typeToFilter);
+    });
+  }
+  
+  private subscribeToFormValueChange(): void {
+    this.form.get('type')!.valueChanges.subscribe((selectedType) => {
+      this.getCategoriesByType(selectedType!);
```

Comment:
There is no need to make a request again after type is changed. I would get all categories, store them in a variable and only filter them later when type is changing.